python/pyarrow/tests/test_array.py

```diff
@@ -310,8 +310,81 @@ def test_array_from_buffers():
     with pytest.raises(TypeError):
         pa.Array.from_buffers(pa.int16(), 3, [u'', u''], offset=1)
 
-    with pytest.raises(NotImplementedError):
-        pa.Array.from_buffers(pa.list_(pa.int16()), 4, [None, values_buf])
+
+def test_string_binary_from_buffers():
+    array = pa.array(["a", None, "b", "c"])
+
+    buffers = array.buffers()
+    copied = pa.StringArray.from_buffers(
```

Comment:
Ah, I hadn't noticed that `StringArray` had a different `from_buffers` method.

Reply:
Frankly I question whether it is useful, I am inclined to remove it

@xhochy

Reply:
Leaving it here for now.

Reply:
Remove this, it is probably only used by `fletcher` and `pyarrow.jvm`.